src/sentry/rules/processing/buffer_processing.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import math
+import uuid
+from abc import ABC, abstractmethod
+from dataclasses import asdict, dataclass
+from datetime import datetime, timezone
+from itertools import islice
+from typing import ClassVar
+
+from celery import Task
+
+from sentry import buffer, options
+from sentry.buffer.redis import BufferHookEvent, redis_buffer_registry
+from sentry.db import models
+from sentry.utils import metrics
+from sentry.utils.registry import NoRegistrationExistsError, Registry
+
+logger = logging.getLogger("sentry.delayed_processing")
+
+
+@dataclass
+class FilterKeys:
+    project_id: int
+
+
+@dataclass
+class BufferHashKeys:
+    model: type[models.Model]
+    filters: FilterKeys
+
+
+class DelayedProcessingBase(ABC):
+    buffer_key: ClassVar[str]
+
+    def __init__(self, project_id: int):
+        self.project_id = project_id
+
+    @property
+    @abstractmethod
+    def hash_args(self) -> BufferHashKeys:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def processing_task(self) -> Task:
+        raise NotImplementedError
+
+
+delayed_processing_registry = Registry[type[DelayedProcessingBase]]()
+
+
+def fetch_alertgroup_to_event_data(
```

Comment:
```suggestion
def fetch_group_to_event_data(
```
nit: since this is generic, should we remove `alert`?